### PR TITLE
[test] test quaternion::uniformRandom

### PR DESCRIPTION
**Pull request by jmirabel**
Related to https://github.com/stack-of-tasks/pinocchio/issues/1146

### CODE REVIEW

File: unittest/quaternion.cpp [hunk 1]
```diff
@@ -28,5 +28,19 @@ BOOST_AUTO_TEST_CASE(test_assignQuaternion)
...
 }
 
+BOOST_AUTO_TEST_CASE(test_uniformRandom)
+{
+  srand(0);
+
+  using namespace pinocchio;
+  Eigen::Quaternion<double> q;
+
+  for (int i = 0; i < 1 << 10; ++i) {
```
Comment (jcarpent) [inline unittest/quaternion.cpp:38]: What is `i < 1 << 10`?

Comment (jmirabel) [reply unittest/quaternion.cpp:38]: I find it more readable than `1 < 10000000000`.

Comment (jmirabel) [reply unittest/quaternion.cpp]: I will change it to `i < (1 << 10)`. I don't know what operator between `<` and `<<` has precedence.

Comment (jcarpent) [reply unittest/quaternion.cpp]: OK. Thanks for the reply.

Comment (jcarpent) [reply unittest/quaternion.cpp]: Next time, you can simply use 1e10.

**Pull request merged**